lib/irb/source_finder.rb

```diff
@@ -26,10 +26,18 @@ def find_source(signature)
       when /\A(?<owner>[A-Z]\w*(::[A-Z]\w*)*)#(?<method>[^ :.]+)\z/ # Class#method
         owner = eval(Regexp.last_match[:owner], context_binding)
         method = Regexp.last_match[:method]
-        if owner.respond_to?(:instance_method)
-          methods = owner.instance_methods + owner.private_instance_methods
-          file, line = owner.instance_method(method).source_location if methods.include?(method.to_sym)
+        return unless owner.respond_to?(:instance_method)
+        if s_count
+          super_owner = owner
+          s_count.times  do |s| 
+            if super_owner.superclass
+              super_owner = super_owner.superclass
+            end
+          end
+          owner = super_owner
         end
+        methods = owner.instance_methods + owner.private_instance_methods
+        file, line = owner.instance_method(method).source_location if methods.include?(method.to_sym)
```

Comment:
Sorry that I only found this now: I think it's also possible to get the method through `super_method`? Something like:

```rb
target_method = owner.instance_method(method)

s_count.times do |s|
  target_method = target_method.super_method if target_method
end

file, line = target_method.source_location if target_method
```